plugins/drill-coverage-plugin/src/adminPartMain/kotlin/Storage.kt

```diff
@@ -1,27 +1,43 @@
 package com.epam.drill.plugins.coverage
 
-import java.util.concurrent.ConcurrentHashMap
+import io.vavr.collection.*
+import io.vavr.kotlin.*
+import kotlinx.atomicfu.*
 
 //TODO This is a temporary storage API. It will be removed when the core API has been developed
 interface StoreKey<T : Any>
 
 interface Storage {
-    suspend fun <T : Any> store(key: StoreKey<T>, value: T)
     suspend fun <T : Any> retrieve(key: StoreKey<T>): T?
+
+    suspend fun <T : Any> store(key: StoreKey<T>, value: T)
+
+    suspend fun <T : Any> update(key: StoreKey<T>, function: (T?) -> T?): T?
+
     suspend fun <T : Any> delete(key: StoreKey<T>): T?
 }
 
 class MapStorage : Storage {
 
-    private val m = ConcurrentHashMap<Any, Any>()
+    private val _map = atomic(HashMap.empty<Any, Any>())
 
     @Suppress("UNCHECKED_CAST")
-    override suspend fun <T : Any> retrieve(key: StoreKey<T>): T? = m[key] as? T
+    override suspend fun <T : Any> retrieve(key: StoreKey<T>): T? = _map.value.getOrNull(key) as T?
```

Comment:
I heard that the right construction for this case is  `value as? T`

Reply:
The idea is to thow a ClassCastException if the type of value is incorrect.